### PR TITLE
Make MArray stay mutable

### PR DESCRIPTION
**Pull request by YingboMa**
Ref: #327

### CODE REVIEW

File: src/abstractarray.jl [hunk 1]
```diff
@@ -78,8 +78,8 @@ similar_type(::Type{A},::Type{T},s::Size{S}) where {A<:AbstractArray,T,S} = defa
 default_similar_type(::Type{T}, s::Size{S}, ::Type{Val{D}}) where {T,S,D} = SArray{Tuple{S...},T,D,prod(s)}
 
 
-# should mutable things stay mutable?
-#similar_type{SA<:Union{MVector,MMatrix,MArray},T,S}(::Type{SA},::Type{T},s::Size{S}) = mutable_similar_type(T,s,length_val(s))
+# should mutable things stay mutable? yes
```
Comment (andyferris) [inline src/abstractarray.jl]: maybe just delete the comment :)

Comment (ChrisRackauckas) [reply src/abstractarray.jl]: I enjoy this.

**Comment by andyferris** [comment]
So... are we going ahead with this?

**Comment by ChrisRackauckas** [comment]
Yes :).

**Comment by YingboMa** [comment]
Yes yes yes :laughing:

**Comment by andyferris** [comment]
:)

And - welcome, Chris!

**Comment by andyferris** [comment]
This will need a breaking release... we're probably not v1.0 yet though.

**Comment by ChrisRackauckas** [comment]
Yup, but minor releases when pre-1.0 are allowed to be breaking in semvar?

**Comment by andyferris** [comment]
Yes!

**Comment by ChrisRackauckas** [comment]
Oh I read what you meant wrong. I thought you meant this would have to wait until StaticArrays v1.0.

**Comment by andyferris** [comment]
Right. That was actually a clumsy attempt to get you to use your new powers and create a release for me ;)

**Pull request merged**